### PR TITLE
fix(dashboard): Create separate file for update

### PR DESCRIPTION
**Pull request by sydopy**
## Overview
Create new file for update dashboard function.

## Tests
[Include a link to the passing GitHub action running the test suite here.]

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/dashboard/src/dashboard-actions/update.spec.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  copyGroup: [],
+  stretchToFit: false,
+  width: 1000,
+  cellSize: 10,
+  undoQueue: [],
+  redoQueue: [],
+  previousPosition: undefined,
+};
+
+it('updates dashboard width', () => {
```
Comment (diehbria) [inline packages/dashboard/src/dashboard-actions/update.spec.ts:19]: i'd like to see a test with recursive merging

**Review by diehbria — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/dashboard-actions/update.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { DashboardState } from '../types';
+import deepMerge from 'deepmerge';
+
+/**
+ * Returns updated dashboard configuration
+ */
+export const update = ({
```
Comment (tjuranek) [inline packages/dashboard/src/dashboard-actions/update.ts:7]: Could we name this something more explicit to what it does? `dashboardUpdate`?

File: packages/dashboard/src/dashboard-actions/update.spec.ts (hunk 1 above)
Comment (tjuranek) [reply packages/dashboard/src/dashboard-actions/update.spec.ts:23]: What's recursive merging?

File: packages/dashboard/src/dashboard-actions/dashboardReducer.ts [hunk 3]
```diff
@@ -170,7 +171,7 @@ export const dashboardReducer: Reducer<DashboardState, DashboardAction> = (
 
     case 'UPDATE':
       return {
-        ...(state = { ...state, ...action.payload.fieldsToUpdate }),
+        ...(state = update({ state, fieldsToUpdate: action.payload.fieldsToUpdate })),
```
Comment (tjuranek) [inline packages/dashboard/src/dashboard-actions/dashboardReducer.ts:174]: We added a lot of code and tests just to perform a deep merge. Why can't we call deep merge here and add a single test for this action? Way less to maintain and I don't see a value add by doing it this way instead.

**Review by tjuranek — CHANGES_REQUESTED** [review]
Seems like this could be simplified by calling deep merge in the action instead of exporting a new update function.

**Comment by tracfren** [comment]
What's the plan for merging this PR in?

**Comment by tjuranek** [comment]
> What's the plan for merging this PR in?

I made the requested changes. Since this pull request was made from a branch off a fork I can't push my changes to it. I'm declining this pull request and re-opened it on a branch off the awslabs/iot-app-kit repo. https://github.com/awslabs/iot-app-kit/pull/214

**Pull request closed without merging**